tango_simlib/tango_yaml_tools/base.py

```diff
@@ -39,57 +39,61 @@ def _build_yaml(self):
+            command_keys.insert(0, command_keys.pop(command_keys.index("name")))
+            command_data = {}
+            for key in command_keys:
+                if key in ["dtype_in", "dtype_out"]:
+                    command_data[key] = command[key].name
+                else:
                     command_data[key] = command[key]
             data_dict[0]["meta"]["commands"].append(command_data)
 
-        for attr in self.parser.get_device_attribute_metadata().values():
-            attr_data = {"name": attr["name"]}
-            if "data_type" in attr:
-                attr_data["data_type"] = attr["data_type"].name
-            if "data_format" in attr:
-                attr_data["data_format"] = attr["data_format"].name
-            if "disp_level" in attr:
-                attr_data["disp_level"] = attr["disp_level"].name
-            # pylint insists on this spacing
-            for key in [
-                    "delta_val",
-                    "period",
-                    "display_unit",
-                    "standard_unit",
-                    "unit",
-                    "max_dim_y",
-                    "max_dim_x",
-                    "label",
-                    "max_value",
-                    "min_alarm",
-                    "max_warning",
-                    "description",
-                    "format",
-                    "delta_t",
-                    "max_alarm",
-                    "min_value",
-                    "inherited",
-                    "min_warning",
-                    "writable",
-                    "writable_attr_name",
-            ]:
-                if key in attr:
+        attr_values = self.parser.get_device_attribute_metadata().values()
+        attr_values = sorted(attr_values, key=lambda x: x["name"])
+        for attr in attr_values:
+            attr_keys = sorted(attr.keys())
+            attr_keys.insert(0, attr_keys.pop(attr_keys.index("name")))
+            attr_data = {}
+            for key in attr_keys:
+                if key in ["data_format", "data_type", "disp_level"]:
+                    attr_data[key] = attr[key].name
+                elif key in [
+                        "name",
+                        "delta_val",
+                        "period",
+                        "display_unit",
+                        "standard_unit",
+                        "unit",
+                        "max_dim_y",
+                        "max_dim_x",
+                        "label",
+                        "max_value",
+                        "min_alarm",
+                        "max_warning",
+                        "description",
+                        "format",
+                        "delta_t",
+                        "max_alarm",
+                        "min_value",
+                        "inherited",
+                        "min_warning",
+                        "writable",
+                        "writable_attr_name",
```

Comment:
I think these keys should be discarded:

- writable_attr_name (result of having a writable attribute)
- inherited (POGO property)
- format  (by-product of the  data_type)
- label (used for GUIs)
- delta_val, delta_t, period (event related, which the values are not specified in the ICD)

Reply:
I'm conflicted, none of the keys are required. If you as a user want to add/drop any  it's up to you.